Formula/typical.rb

```diff
@@ -0,0 +1,44 @@
+class Typical < Formula
+  desc "Data interchange with algebraic data types"
+  homepage "https://github.com/stepchowfun/typical"
+  url "https://github.com/stepchowfun/typical/archive/v0.9.5.tar.gz"
+  sha256 "9f6895454d97c0c3b0e44e0bfe0e68a6558b8ac8c06a52ee6a04c37b4932173e"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    (testpath/"types.t").write <<~EOS
+      struct SendEmailRequest {
+          to: String = 0
+          subject: String = 1
+          body: String = 2
+      }
+
+      choice SendEmailResponse {
+          success = 0
+          error: String = 1
+      }
+    EOS
+
+    assert_match "", shell_output("#{bin}/typical generate types.t --rust types.rs --typescript types.ts")
```

Comment:
actually we should do this
```suggestion
    assert_empty shell_output("#{bin}/typical generate types.t --rust types.rs --typescript types.ts")
```